src/App.tsx

```diff
@@ -46,6 +46,7 @@ LogBox.ignoreLogs([
   // of TabNavigator. As of today, no bug has been observed which seems related to the warning.
   'Cannot update a component',
   'EventEmitter.removeListener',
+  'Non-serializable values were found in the navigation state',
```

Comment:
Je l'ai ajouté pour éviter le warning : 
```
Non-serializable values were found in the navigation state. Check:

TabNavigator > params.params.date.selectedDate (Mon Oct 31 2022 12:09:10 GMT+0100 (heure normale d’Europe centrale))

This can break usage such as persisting and restoring state. This might happen if you passed non-serializable values such as function, class instances etc. in params. If you need to use components with callbacks in your options, you can use 'navigation.setOptions' instead. See https://reactnavigation.org/docs/troubleshooting#i-get-the-warning-non-serializable-values-were-found-in-the-navigation-state for more details.
```
Lorsque le navigate est exécuté ligne 149 dans `DatesHoursModal.tsx` par exemple la date à un type `Date` et à l'arrivée dans l'URL on se retrouve avec un type `string` au format  `2022-10-31T11:09:10.411Z` sans que je ne comprenne comment.
Si on veut se passer du retrait du warning via la LogBox il faudrait avoir un type `string` systématiquement et régler les effets de bords qui vont arriver ailleurs.
Preneur de ton avis :)

Reply:
Il faudrait sérialiser la date
> $ new Date()
> Mon Oct 31 2022 15:02:26 GMT+0200 (Eastern European Standard Time)

parse et stringify

À tu tester avec ça  src/features/navigation/screenParamsUtils.ts ?

- screenParamsParser
- screenParamsStringifier

On est pas censé avoir des choses non sérialisable/déserialisable dans l'URL et les navigations screens params sont dans l'URL
Donc je dirais de supprimer la suppression du warning (qui est légitime) et traiter les query params avec ça

Reply:
En effet, l'URL ne devrait contenir que des strings. Il faut pouvoir traiter en amont et en aval, c'est-à-dire que les paramètres dans l'URL devraient être sérialisés en string, et lors de la récupération devraient être parsés.

C'est plus délicat avec les `array` et `hash` mais d'une part c'est partiellement traité par JavaScript, d'autre part ce n'est pas le sujet :P

Reply:
Plus concrètement dans le `SearchState` tu ne devrais pas manipuler de `Date` mais un string. Parce que c'est la même chose : lorsque tu fais une requête à une API, il faut passer des string